### ISSUE TITLE
Dynamic reloading of new targets, for example ecs container ips, or a targets/refresh endpoint

### ISSUE THREAD

**Issue by rjshrjndrn** [grafana/phlare#446]
#### Is your feature request related to a problem? Please describe.

Hi,

When we're monitoring ecs containers using phlare, we can create file-based scrape config, but if there is a change in the container IP, How to inform phlare about that change?

In Prometheus we've file_sd_configs and Prometheus will refresh the targets if and when there's a change. Does phlare offer something similar?

Or

Is there any endpoint for reloading the scrape targets?

#### Describe the solution you'd like

Either implementation of prometheus `file_sd_configs` or a `v1/targets/refresh` endpoint for getting new targets from the scrape config

#### Describe alternatives you've considered

restart phlare every few minutes

**Comment by simonswine** [comment]
You are right currently Phlare only supports the Kubernetes and the Static discovery and on top of that we don't support config reloads.

Our plan for the next release is to actually "out-source" the scraping/collection of profiles to Grafana Agent (https://github.com/grafana/agent). Then I would think we would support all the discovery method you know from Prometheus. In Phlare we would then only have those two basic discovery method of Kubernetes and Statuc. 

Would that work for you?

**Comment by rjshrjndrn** [comment]
Hi @simonswine Thank you for the update. Sounds good to have the integration. 

Grafana Agent is a great tool to have a centralized scrapping mechanism for large enterprises.
But don't you think, it'll be overkill to run, manage and learn one more instance just for the scraping?

Can't we directly plug in the Prometheus discovery module?
https://github.com/prometheus/prometheus/tree/main/discovery

**Comment by rjshrjndrn** [comment]
Now, thats a community project. Thank you very much @simonswine for being open to suggestions. Eagerly waiting for the next release.
